src/translate.scala

```diff
@@ -12,68 +13,105 @@ import lambdapi._
+      if (cut.length == 1) ("", cut(0))
+      else (cut(0), cut(1).replace(".", "__"))
+
+    var new_id = radical
+    if (namesSet(new_id))
+      new_id += "__" + kind
+    if (namesSet(new_id))
+      new_id = prefix + "__" + new_id
+    if (namesSet(new_id))
+      error("duplicate name")
+
+    namesSet += new_id
+    namesMap += full_name(id, kind) -> new_id
+    new_id
+  }
+
+  def name_get(a: String, kind: String): String = {
+    namesMap get full_name(a, kind) match {
+      case None => names_add(a, kind)
+      case Some(s) => s
+    }
+  }
 
-  val typeId = "Typ"
-  val  etaId = "eta"
-  val  epsId = "eps"
+  /* kinds */
 
-  val TypeT = Syntax.Symb(typeId)
-  val  etaT = Syntax.Symb( etaId)
-  val  epsT = Syntax.Symb( epsId)
+  // def kind(a: String, k: Export_Theory.Kind.Value): String = a + "|" + k.toString
 
+  def class_ident(a: String): String = name_get(a, Export_Theory.Kind.CLASS.toString)
+  def  type_ident(a: String): String = name_get(a, Export_Theory.Kind.TYPE .toString)
+  def const_ident(a: String): String = name_get(a, Export_Theory.Kind.CONST.toString)
+  def axiom_ident(a: String): String = name_get(a, Export_Theory.Kind.AXIOM.toString)
+  def   thm_ident(a: String): String = name_get(a, Export_Theory.Kind.THM  .toString)
+  def   var_ident(a: String): String = name_get(a, "var")
 
-  /* kinds */
+  def proof_ident(serial: Long): String = "proof" + serial
 
-  def kind(a: String, k: Export_Theory.Kind.Value): String = a + "|" + k.toString
+  /* special names */
 
-  def class_kind(a: String): String = kind(a, Export_Theory.Kind.CLASS)
-  def type_kind(a: String): String = kind(a, Export_Theory.Kind.TYPE)
-  def const_kind(a: String): String = kind(a, Export_Theory.Kind.CONST)
-  def axiom_kind(a: String): String = kind(a, Export_Theory.Kind.AXIOM)
-  def thm_kind(a: String): String = kind(a, Export_Theory.Kind.THM)
+  val typeId: String = const_ident("Typ")
+  val  etaId: String = const_ident("eta")
+  val  epsId: String = const_ident("eps")
 
-  def proof_kind(serial: Long): String = "proof" + serial
+  val TypeT: Syntax.Term = Syntax.Symb(typeId)
+  val  etaT: Syntax.Term = Syntax.Symb( etaId)
+  val  epsT: Syntax.Term = Syntax.Symb( epsId)
 
 
-  val typeD = Syntax.Declaration(typeId, Nil, Syntax.TYPE)
-  val  etaD = Syntax.Declaration(etaId, Nil, Syntax.arrow(TypeT, Syntax.TYPE), const = false)
+  val typeD: Syntax.Command  = Syntax.Declaration(typeId, Nil, Syntax.TYPE)
+  val  etaN: Syntax.Notation = Syntax.Prefix("η", 10)
+  val  etaD: Syntax.Command  = Syntax.DefableDecl(etaId, Syntax.arrow(TypeT, Syntax.TYPE), inj = true, not = Some(etaN))
 
 
+  val epsN: Syntax.Notation = Syntax.Prefix("ε", 10)
+  val epsD: Syntax.Command =
+  {
+    val prop = Syntax.Symb(type_ident(Pure_Thy.PROP))
+    val eta_prop = Syntax.Appl(etaT, prop)
+    Syntax.DefableDecl(epsId, Syntax.arrow(eta_prop, Syntax.TYPE), not = Some(epsN))
+  }
+
+  // Integration rewrites
+
   /* produces "rule f (g &a &b) → f &a ⇒ f &b */
```

Comment:
Thanks to replace f and g by etaeps and funimp to better understand the comment.